SofaKernel/modules/SofaDeformable/PolynomialSpringsForceField.inl

```diff
@@ -0,0 +1,598 @@
+                    mat->add(offset + m_dimension * secondIndex + i, offset + m_dimension * firstIndex + j, stiffnessDeriv);
+                    mat->add(offset + m_dimension * secondIndex + i, offset + m_dimension * secondIndex + j, -stiffnessDeriv);
+                }
+            }
+        }
+    } else {
+        sofa::core::behavior::MultiMatrixAccessor::MatrixRef mref11 = matrix->getMatrix(this->mstate1);
+        sofa::core::behavior::MultiMatrixAccessor::MatrixRef mref22 = matrix->getMatrix(this->mstate2);
+        sofa::core::behavior::MultiMatrixAccessor::InteractionMatrixRef mref12 = matrix->getMatrix(this->mstate1, this->mstate2);
+        sofa::core::behavior::MultiMatrixAccessor::InteractionMatrixRef mref21 = matrix->getMatrix(this->mstate2, this->mstate1);
+        if (!mref11 && !mref22 && !mref12 && !mref21) return;
+
+        for (unsigned int index = 0; index < m_firstObjectIndices.size(); index++)
+        {
+            firstIndex = m_firstObjectIndices[index];
+            secondIndex = m_secondObjectIndices[index];
+            const JacobianMatrix& jacobMatrix = m_differential[index];
+
+            bool contains = false;
+            for (unsigned s = 0; s < addSubIndex.size() && !contains; s++) if (firstIndex == addSubIndex[s]) contains = true;
+            if (!contains) continue;
+
+            for(unsigned int i = 0; i < m_dimension; i++)
+            {
+                for (unsigned int j = 0; j < m_dimension; j++)
+                {
+                    Real stiffnessDeriv = jacobMatrix[i][j] * kFact;
+                    mref11.matrix->add(mref11.offset + m_dimension * firstIndex + i, mref11.offset + m_dimension * firstIndex + j, -stiffnessDeriv);
+                    mref12.matrix->add(mref12.offRow + m_dimension * firstIndex + i, mref12.offCol + m_dimension * secondIndex + j, stiffnessDeriv);
+                    mref21.matrix->add(mref21.offRow + m_dimension * secondIndex + i, mref21.offCol + m_dimension * firstIndex + j, stiffnessDeriv);
+                    mref22.matrix->add(mref22.offset + m_dimension * secondIndex + i, mref22.offset + m_dimension * secondIndex + j, -stiffnessDeriv);
+                }
+            }
+        }
+    }
+}
+
+
+
+template<class DataTypes>
+double PolynomialSpringsForceField<DataTypes>::PolynomialValue(unsigned int springIndex, double strainValue)
+{
+    helper::ReadAccessor<Data<VecReal>> vPolynomialStiffness = d_polynomialStiffness;
+    helper::ReadAccessor<Data<helper::vector<unsigned int> >> vPolynomialDegree = d_polynomialDegree;
+
+    msg_info() << "Polynomial data: ";
+    double highOrderStrain = 1.0;
+    double result = 0.0;
+    for (size_t degreeIndex = 0; degreeIndex < vPolynomialDegree[springIndex]; degreeIndex++) {
+        highOrderStrain *= strainValue;
+        result += vPolynomialStiffness[m_polynomialsMap[springIndex][degreeIndex]] * highOrderStrain;
+        msg_info() << "Degree:" << (degreeIndex + 1) << ", result: " << result;
```

Comment:
Don't do msg_info() in a for loop.... it is better to have a single msg_info after the loop...and concatenating all the "degree" into a single stringstream. Both faster and clearer.

If the aim of this loop is to print something then put everything in a if(notMuted()){}